### PR TITLE
[SURGE-3012] Create new base theme

### PR DESCRIPTION
**Pull request by jordanpagewhite**
### Github Issue Link
* #3012 

### Verification Process

* All build steps are green
* The rhdp2 theme is installed and is the default theme
* The rhdp theme is uninstalled
* The rhdp2 theme is referencing redhat-developer/rhd-frontend as a git submodule
* Some of the dist assets in the redhat-developer/rhd-frontend project are referenced in the rhdp2.libraries.yml file so that we can demonstrate that they will load properly in Drupal
* A minimal (minimal is relative in this case) number of Twig templates are brought over from the `rhdp` theme to make fundamental pieces of the site _work_ like the header, footer, html.html.twig, page.html.twig, etc.

### CODE REVIEW

**Comment by robpblake** [comment]
@jordanpagewhite Drupal isn't booting because it is complaining that the `rhdp` theme will not be installed after `drush cim` and that a whole bunch of other stuff depends on that theme being installed.

**Comment by jordanpagewhite** [comment]
Ah, yeah, of course. Thanks Rob. I pushed this up towards the end of my day yesterday, and didn't have a chance to circle back after the build processes. I will take care of that sometime this morning.

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
This is failing because we cannot successfully pull the redhat-theme git submodule

File: _docker/drupal/Dockerfile.mp [hunk 1]
```diff
@@ -24,12 +24,19 @@ COPY drupal-filesystem/web/sites/default /var/www/drupal/web/sites/default
...
 COPY drupal-filesystem/web/themes/custom /var/www/drupal/web/themes/custom
 
-# Build the theme
+# Build the rhdp theme
 RUN cd /var/www/drupal/web/themes/custom/rhdp/rhd-frontend \
     && npm install \
     && npm run-script build \
     && rm -rf /var/www/drupal/web/themes/custom/rhdp/rhd-frontend/node_modules
 
+# Build the rhdp2 theme
```
Comment (jordanpagewhite) [inline _docker/drupal/Dockerfile.mp]: @robpblake This is the PR, and the specific part that fails, that we discussed today. If I uncomment this block/command and execute the build, it completes successfully. I then have to pull the rhd-frontend git module, cd into the directory, run `npm install`, run `npm run styles`, run `npm run build` and then I have the `dist` directory with the dist assets as intended. This RUN command will fail though.

We need some way to make this RUN command execute successfully or some command that yields our desired end state of the dist assets being built out in the `dist` directory within `rhd-frontend`. I think that one thing, and there may be others, that current causes this to fail is the new method of including the FortAwesome depenendenc(ies). @KamiQuasi suggested the following:

> we would just commit the .npmrc file with this:
```@fortawesome:registry=https://npm.fontawesome.com/
//npm.fontawesome.com/:_authToken=${FONTAWESOME_NPM_AUTH_TOKEN}```

where `${FONTAWESOME_NPM_AUTH_TOKEN}` is an env variable.

Comment (robpblake) [reply _docker/drupal/Dockerfile.mp]: @jordanpagewhite As discussed yesterday, I'll work on a temporary solution that lets us build this as part of the Docker build until the CDN work is ready. Please keep the CDN stuff in the back of your mind when doing the integration with the new `rhd-frontend`. 

Standby for a PR to fix this issue.

**Comment by jordanpagewhite** [comment]
The build process must fail to successfully execute `npm install` within `./web/themes/custom/rhdp2/rhd-frontend`. There is no `node_modules` directory in the container, and the `dist` directory is also missing.

**Comment by jordanpagewhite** [comment]
I did not observe any errors in the build though

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by robpblake** [comment]
@jordanpagewhite I'll need to add the FONTAWESOME_LICENCE_KEY argument to our MP builds as well. I forgot to do that on Friday.

Sorry about that.

cheers,

Rob

**Comment by robpblake** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
@jordanpagewhite The glone from  Git Hub is failing with the following error:

```

Cloning "https://github.com/redhat-developer/developers.redhat.com " ...

| error: No submodule mapping found in .gitmodules for path 'src/styles/redhat-theme'
| Failed to recurse into submodule path '_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend'

```

**Comment by jordanpagewhite** [comment]
I've created this PR against the rhd-frontend project to try to resolve the error posted by @robpblake above: https://github.com/redhat-developer/rhd-frontend/pull/132

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Pull request merged**